### PR TITLE
(python3-streams) fix AU script

### LINKED ISSUE #2555: (python) Outdated, build failing with null
### Checklist

- [X] I have checked the moderation queue that no new version has been submitted (only visible when logged in), or the submitted package version has failed the automated checks.

Please navigate to the following [link](https://community.chocolatey.org/packages?q=&moderatorQueue=true&moderationStatus=all-statuses&prerelease=false&sortOrder=package-download-count) to view the moderation queue.

- [X] I have verified that the new version is a stable release.
- [X] I have looked to see if there are any Bug reports that prevents a new version being submitted and that no existing Outdated Reports have been created.
- [X] I have verified the new version has a Windows binary (EXE or MSI installer, Zip or other archive, etc).
- [X] I have verified that this is the correct repository, and the package is maintained by the chocolatey-community user.

### New Software Version

3.12.7, 3.11.10

### Download location

_No response_

### Package Page

_No response_

### PR DESCRIPTION
**Pull request by TheCakeIsNaOH**
## Description

The zip directory naming is inconsistent, this directly uses the filenames to find the appropriate license and copyright files.

## Motivation and Context

Fixes #2555

## How Has this Been Tested?

Tested on Win10 and in test env

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:

- [x] My code follows the code style of this repository.
- [ ] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [ ] I have updated the package description and it is less than 4000 characters.
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey-community/chocolatey-packages/blob/master/CONTRIBUTING.md)
- [x] The added/modified package passed install/uninstall in the Chocolatey Test Environment(https://github.com/chocolatey-community/chocolatey-test-environment/). _Note that we don't support the use of any other environment_.
- [x] The changes only affect a single package (not including meta package).

### CODE REVIEW

**Comment by sigma67** [comment]
Is there any ETA on this PR? Since there has been no activity in 10 days

**Comment by sigma67** [comment]
@JPRuskin could you kindly review? it would be nice to have a package for 3.13 almost 2 months after its release

**Comment by rob-neobim** [comment]
@JPRuskin @sigma67 Just a +1 here. I'm not sure there's anything I can do to help except vote this PR up.

**Comment by rob-neobim** [comment]
@AdmiringWorm Perhaps you could review this?

**Comment by pauby** [comment]
@rob-neobim @sigma67 Please don't ping people on PR's. The community team will get to it whentime allows.

**Comment by yan12125** [comment]
Thanks! I can confirm revised update.ps1 works for 3.13.1. However, there is another error related to zip files for 3.14.0a2:
```
Downloading to python-3.14.0a2.exe - https://www.python.org/ftp/python/3.14.0/python-3.14.0a2.exe
Downloading to python-3.14.0a2-amd64.exe - https://www.python.org/ftp/python/3.14.0/python-3.14.0a2-amd64.exe
MethodInvocationException: C:\Users\yen\Downloads\chocolatey-packages\automatic\python3-streams\update.ps1:41
Line |
  41 |    $response_stream = $webrequest.GetResponse().GetResponseStream()
     |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
```
In https://docs.python.org/3.14/archives/, there is `python-3.14.0a0-docs-text.zip` but no `python-3.14.0a2-docs-text.zip`. I'm not sure if missing files are caused by some unexpected errors or intended changes. If the latter is the case, this pull request may need major revisions.

**Comment by rob-neobim** [comment]
Thanks for checking, @yan12125! I attempted to check this out, but there are no instructions on how to test anything. Everything I try falls over with an exception `An exception ocurred during Chocolatey AU update. Cancelling all other checks.` or `The term 'updateall' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.` Could someone who knows what they're doing explain how to test this particular package?

**Comment by TheCakeIsNaOH** [comment]
From https://github.com/python/pythondotorg/issues/2672
> To summarise, there are no official docs built for alpha and beta pre-releases. Release candidates and full releases do come with official docs builds.

Looks like I"ll have to update this PR to take that into account.


> I attempted to check this out, but there are no instructions on how to test anything.

This package is an automatic package, in that the package version (and other associated files) is updated by a CI build that uses the `Chocolatey-AU` module with the `update.ps1`. I thought there was an explanation somewhere on how to setup and test the update scripts, but the module documentation itself is the best I can find:
https://github.com/chocolatey-community/chocolatey-au/

In short, you will need to have the Chocolatey-AU module installed and imported into in powershell, and then run the `update.ps1` script. 
I think this package may also needs some addtional helper scripts imported as well: https://github.com/chocolatey-community/chocolatey-packages/tree/master/scripts

**Comment by robmoss2k** [comment]
Thanks, @TheCakeIsNaOH. That's really helpful information, and it should probably be included somewhere in this repository. I'll have a tinker at some point next week.

**Comment by yan12125** [comment]
> Looks like I"ll have to update this PR to take that into account.

Are txt files required here? Otherwise the following files may be an option:
https://github.com/python/cpython/blob/main/Doc/copyright.rst
https://github.com/python/cpython/blob/main/Doc/license.rst

If I understand CPython's [Makefile](https://github.com/python/cpython/blob/main/Doc/Makefile) correctly, txt files are generated from corresponding rst files.

**Comment by robmoss2k** [comment]
I think the license is simple enough, as that's currently just a web link: [https://docs.python.org/3.13/license.html](https://docs.python.org/3.13/license.html)

Copyright is somewhat trickier, but we can use a regex to (badly) strip the RST stuff, preamble, and postamble:

```ps1
((Invoke-WebRequest -Uri 'https://raw.githubusercontent.com/python/cpython/refs/heads/main/Doc/copyright.rst').Content) -replace "\n", " " -replace "\*+ Copyright \*+  Python and this documentation is:  " -replace "  -------.*$"
```

**Comment by robmoss2k** [comment]
I've submitted a replacement PR #2594.

**Comment by robmoss2k** [comment]
I also raised an [issue](https://github.com/chocolatey-community/chocolatey-au/issues/73) upstream, noting that Appveyor is testing the wrong version of Python.

**Comment by TheCakeIsNaOH** [comment]
> I also raised an https://github.com/chocolatey-community/chocolatey-au/issues/73 upstream, noting that Appveyor is testing the wrong version of Python.

The fix for that would be to return an ordered dictionary of the steams (in the correct versioning order), instead of returning a hashtable.

**Comment by TheCakeIsNaOH** [comment]
Superseeded by #2594
